src/firefly/js/charts/ui/PlotlyWrapper.jsx

```diff
@@ -166,24 +167,58 @@ export class PlotlyWrapper extends Component {
         return true;
     }
 
+    optimize(graphDiv, renderType, {chartId, data, layout, dataUpdate, layoutUpdate, dataUpdateTraces, ...rest}) {
+        const {lastInputTime} = layout;
+        if (lastInputTime && renderType ===  RenderType.NEW_PLOT && graphDiv.data) {
+            // omitting 'firefly' from data[*] for now
+            data = data.map((d) => omit(d, 'firefly'));
```

Comment:
I think the mismatch in clicked and highlighted point when highlighting a selected point is caused not passing firefly rowIdx array to plotly div. (For selected points I get row idx from plotly div, because we do not save them in data.) I think, we should omit firefly object for the purposes of calculating the difference, but it should not be dropped completely. We should still pass it with the partial or full update.